src/client_side.cc

```diff
@@ -3007,7 +3007,11 @@ ConnStateData::proxyProtocolError(const char *msg)
 static const SBuf Proxy1p0magic("PROXY ", 6);
 
 /// magic octet prefix for PROXY protocol version 2
-static const SBuf Proxy2p0magic("\x0D\x0A\x0D\x0A\x00\x0D\x0A\x51\x55\x49\x54\x0A", 12);
+//  The first four bytes (\x0D\x0A\x0D\x0A) are intentionally
+//  ommited from the prefix since leading whitespace is stripped
```

Comment:
The stripped leading whitespace may differ from the intentionally omitted bytes. Thus, this change will result in non-PROXY prefixes treated as PROXY prefixes. Such relaxed treatment of _framing-related_ constructs is undesirable. Can this change be adjusted to check stripped bytes and reject non-PROXY prefixes correctly?

Reply:
I'm not especially familiar with the SBuf class - are consumed bytes immediately freed, or is there a way to look back? Or at least to check the current position in the stream?

I can spend some more time on this later in the week, but any pointers would be appreciated.

Reply:
No, SBuf is somewhat similar to std::string -- not suitable for lookback operations.

If Amos response is not enough, consider not consuming all whitespace until there is something other than whitespace present in the buffer or it becomes clear that the whitespace prefix is not going to match.